### DISCUSSION TITLE
[Request] - Allow for configuration to prevent workflow runs from being deleted from GitHub Actions

### DISCUSSION THREAD

**Discussion by kjluedke** [community/community#7810] [Actions] [answered]
By default, users can delete workflow runs if the run is completed or more than two weeks old (https://docs.github.com/en/actions/managing-workflow-runs/deleting-a-workflow-run)

I wish to have a configuration that forbids this deletion so that an auditable log is maintained.  Currently, a user could trigger an action (a deployment, for example), and could delete the log entry to eliminate evidence that the action was ever triggered.  An immutable log would prevent this.

**Comment by ethomson** [comment, marked as answer]
Thanks for the feedback, this makes sense.  We'll add this to the backlog.

**Reply by syerneni-sp** [reply]
Is this feature added on github now? I wasnt able to find any reference in documentation

**Reply by Power-Maverick** [reply]
Any estimate on when this will be released?

**Reply by rubenbezos** [reply]
This feature is a must. Workflow runs deletion should be restricted at an enterprise and/or organization level.

**Reply by thundering-herd** [reply]
It looks like @ethomson is no longer at GitHub. So the question is, who picked this up afterwards?

**Reply by anroots-pm** [reply]
Was this ever added to internal GitHub backlog?

Either an immutability toggle; or permissions config (limited to certain org roles) would solve this, and give some kind of auditable proofing that the log can't easily be tampered with.

**Reply by jsimerly-lilly** [reply]
Any update on this?
